lib/assert.js

```diff
@@ -79,6 +79,10 @@ function getMessage(self) {
 // display purposes.
 
 function fail(actual, expected, message, operator, stackStartFunction) {
+  if (arguments.length === 1)
+    message = actual;
+  if (arguments.length === 2)
+    operator = '!=';
```

Comment:
There are at least three issues with that.

First, that introduces additional behavior changes I was hoping to avoid.

Current behavior preserved in this PR:

```
> assert.fail()
AssertionError: undefined undefined undefined
```

With the default parameters change:

```
> assert.fail()
AssertionError: undefined != undefined
```

Second, it breaks the two-argument feature added here.

Current PR:

```
> assert.fail('first', 'second');
AssertionError: 'first' != 'second'
```

With the suggested change:

```
> assert.fail('first', 'second')
AssertionError: first
```

Third, and most importantly, it breaks one of the two current usable argument combinations for the function.

Current behavior also preserved in this PR:

```
> assert.fail('first', 'second', undefined, 'operator')
AssertionError: 'first' operator 'second'
```

With the default parameters as suggested replacing the `arguments.length` checks:

```
> assert.fail('first', 'second', undefined, 'operator')
AssertionError: first
```

There's no doubt a way around this, and maybe the problem is my lack of familiarity with default parameters?